src/modules/launcher/Plugins/Microsoft.Plugin.Indexer/Main.cs

```diff
@@ -95,7 +95,14 @@ public List<Result> Query(Query query)
                             });
                         }
 
-                        var searchResultsList = _api.Search(searchQuery, maxCount: _settings.MaxSearchCount).ToList();
+                        var searchResultsList = _api.Search(searchQuery, isFullQuery, maxCount: _settings.MaxSearchCount).ToList();
+
+                        // If the delayed execution query is not required (since the SQL query is fast) return empty results
+                        if (searchResultsList.Count == 0 && isFullQuery)

```

Comment:
[nit] I think we can remove this because if `searchResultsList` is empty, then the for loop ahead would not be executed. The warning would end up being shown but that seems to be the current default behaviour. @alekhyareddy28 can tell more on this.

Reply:
I had to add this because it could either return 0 or 1 element depending on the Drive detection warning. Since in the QueryResults method I wouldn't have a generic way of knowing if it contains the drive detection warning or not this was the best solution.